sound/soc/sof/loader.c

```diff
@@ -129,6 +130,104 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+static ssize_t snd_sof_ext_man_size(const struct firmware *fw)
+{
+	const struct sof_ext_man_header *head = (void *)fw->data;
```

Comment:
would be much better to cast to the target type, i.e. `(const struct sof_ext_man_header *)fw->data`

Reply:
Is there any difference?

Reply:
I think so, yes. Type-casting should be avoided whenever possible and where not possible, done in the most appropriate and careful way.